src/main/scala/tilelink/Parameters.scala

```diff
@@ -1085,27 +1151,58 @@ case class TLEdgeParameters(
   def manager = slave
   def client = master
 
-  val maxTransfer = max(client.maxTransfer, manager.maxTransfer)
+  val maxTransfer = max(master.maxTransfer, slave.maxTransfer)
   val maxLgSize = log2Ceil(maxTransfer)
 
   // Sanity check the link...
-  require (maxTransfer >= manager.beatBytes, s"Link's max transfer (${maxTransfer}) < ${manager.managers.map(_.name)}'s beatBytes (${manager.beatBytes})")
-
-  val bundle = TLBundleParameters(client, manager)
-  def formatEdge = client.infoString + "\n" + manager.infoString
+  require (maxTransfer >= slave.beatBytes, s"Link's max transfer (${maxTransfer}) < ${slave.slaves.map(_.name)}'s beatBytes (${slave.beatBytes})")
+
+  // For emits, check that the source is allowed to send this transactions
+// TODO emitAcquire, source => sourceId (or don't need sourceId at all)
+//      Probe, source => address
```

Comment:
These A channel messages from MasterToSlave are:

being routed to a slave based on a address, so they need to be passed to a helper that checks whether the `slave` that owns certain `addresses` claimed to `support` transactions of this size/type
being sent from the client using a sourceId, so they need to be passed to a helper that checks with the `master` that owns certain `sourceIds` claimed to `emit` transactions of this size/type